mantidproject/mantid
# Indirect - Data Reduction - Disable run, plot and save options while processing

**Description of work.**
This PR moves the Run button on the Data Reduction interface so that it is above the output options in order to remain consistent with other interfaces.
This PR also ensures the output options (Plot and Save) are disabled while the interface is running, as well as the Run button.
The Plot, Save and Run buttons are also now disabled while plotting is taking place.

**To test:**
1. `Interfaces`->`Indirect`->`Data Reduction`
2. Check that the Run button has been moved to above the output options on all of the tabs
3. Enter the data below
4. Click `Run` and make sure that Run, Plot and Save options are all disabled and then re-enabled once the running is complete.
5. Click `Plot` and make sure the Run, Plot and Save options are all disabled and then re-enabled once the plotting is complete

**Test Data**

**ISISEnergyTransfer:**
Use run-number `26176`
**ISISCalibration:**
Use run-number `26176`
**ISISDiagnostics:**
Use run-number `26176`
**Transmission**
Sample: run number `26176`
Container: run number `26174`
**Symmetrise**
Input File: [irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2486896/irs26176_graphite002_red.zip)
Move the vertical lines on the graph slightly towards 0
Click `Preview`, then click `Run`
**S(Q,w)**
Input File: [irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2486896/irs26176_graphite002_red.zip)
QLow: 0.45                      
QWidth: 0.05                     
QHigh: 1.6
ELow: -0.5                      
EWidth: 0.005                     
EHigh: 0.5
Note that the run button is disabled for this tab when running, however it may appear that it isn't.
**Moments**
Input File: [iris26176_graphite002_sqw.zip](https://github.com/mantidproject/mantid/files/2486919/iris26176_graphite002_sqw.zip)

Fixes #23774 

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
